verification/src/solidity/compiler_cache.rs

```diff
@@ -0,0 +1,85 @@
+use ethers_solc::Solc;
+use semver::Version;
+use std::{collections::HashMap, sync::Arc};
+
+#[derive(Default)]
+pub struct CompilerCache {
+    cache: std::sync::Mutex<HashMap<Version, Arc<tokio::sync::RwLock<Option<Solc>>>>>,
+}
+
+impl CompilerCache {
+    async fn try_get(&self, ver: &Version) -> Option<Solc> {
+        let entry = {
+            let cache = self.cache.lock().unwrap();
+            cache.get(ver).cloned()
+        };
+        match entry {
+            Some(lock) => {
+                let solc = lock.read().await;
+                solc.as_ref().cloned()
+            }
+            None => None,
+        }
+    }
+
+    async fn fetch(&self, ver: &Version) -> anyhow::Result<Solc> {
+        let lock = {
+            let mut cache = self.cache.lock().unwrap();
+            Arc::clone(cache.entry(ver.clone()).or_default())
+        };
+        let mut entry = lock.write().await;
+        match entry.as_ref() {
+            Some(solc) => Ok(solc.clone()),
+            None => {
+                log::info!("installing solc version {}", ver);
```

Comment:
```suggestion
                log::info!(target: "compiler_cache", "installing solc version {}", ver);
```

Consider using `target` key describing the module which generated a log